packages/flutter_tools/templates/module/ios/library/Flutter.tmpl/podhelper.rb.tmpl

```diff
@@ -110,37 +123,30 @@ def install_flutter_application_pod(flutter_application_path)
     :execution_position => :before_compile
 end
 
-def parse_KV_file(file, separator='=')
-  file_abs_path = File.expand_path(file)
-  if !File.exists? file_abs_path
-    return [];
-  end
-  pods_array = []
-  skip_line_start_symbols = ["#", "/"]
-  File.foreach(file_abs_path) { |line|
-    next if skip_line_start_symbols.any? { |symbol| line =~ /^\s*#{symbol}/ }
-    plugin = line.split(pattern=separator)
-    if plugin.length == 2
-      podname = plugin[0].strip()
-      path = plugin[1].strip()
-      podpath = File.expand_path("#{path}", file_abs_path)
-      pods_array.push({:name => podname, :path => podpath});
-     else
-      puts "Invalid plugin specification: #{line}"
-    end
-  }
-  return pods_array
+# .flutter-plugins-dependencies format documented at
+# https://flutter.dev/go/plugins-list-migration
+def flutter_parse_plugins_file(file)
+  file_path = File.expand_path(file)
+  return [] unless File.exists? file_path
+
+  dependencies_file = File.read(file)
+  dependencies_hash = JSON.parse(dependencies_file)
+
+  # dependencies_hash.dig('plugins', 'ios') not available until Ruby 2.3
+  return [] unless dependencies_hash.has_key?('plugins')
+  return [] unless dependencies_hash['plugins'].has_key?('ios')
+  dependencies_hash['plugins']['ios'] || []
 end
 
 def flutter_root
-  generated_xcode_build_settings = parse_KV_file(File.join(__dir__, 'Generated.xcconfig'))
-  if generated_xcode_build_settings.empty?
-    puts "Generated.xcconfig must exist. Make sure `flutter pub get` is executed in the Flutter module."
-    exit
+  generated_xcode_build_settings_path = File.expand_path(File.join('..', '..', 'Flutter', 'Generated.xcconfig'), __FILE__)
+  unless File.exist?(generated_xcode_build_settings_path)
+    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
   end
-  generated_xcode_build_settings.map { |p|
-    if p[:name] == 'FLUTTER_ROOT'
-      return p[:path]
-    end
-  }
+
+  File.foreach(generated_xcode_build_settings_path) do |line|
+    matches = line.match(/FLUTTER_ROOT\=(.*)/)
+    return matches[1].strip if matches
+  end
+  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
```

Comment:
add a code comment or reference issue for when this can happen?

Reply:
I don't think it can ever happen AFAIK, I think I just wanted a way to surface Something Bad Happened and stopping instead of copying from a nonexistent directory in https://github.com/flutter/flutter/pull/59044/files#diff-76053fa099530bf75a339336cccbc2efR23.